### PR TITLE
Prac 05 code review request

### PR DESCRIPTION
**Pull request by Dav0-12**
@Luke-Hawkins. Thanks Luke : )

### CODE REVIEW

File: prac_05/state_names.py [hunk 1]
```diff
@@ -8,10 +8,16 @@
...
+    try:
         print(state_code, "is", CODE_TO_NAME[state_code])
-    else:
+    except KeyError:
         print("Invalid short state")
     state_code = input("Enter short state: ")
+
+
+max_code_length = max(len(code) for code in CODE_TO_NAME.keys())
+max_name_length = max(len(name) for name in CODE_TO_NAME.values())
```
Comment (Luke-hawkins) [inline prac_05/state_names.py]: Looks pretty good, but I'm not sure if you need to find the max_name_length for the values as it doesn't affect the formatting, I think would just start printing from the "is" anyway :).

File: prac_05/hex_colours.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+"""
+CP1404/CP5632 Practical
+Hex Colours
+"""
+
+COLOUR_TO_HEX = {"aliceblue": "#f0f8ff", "absolute zero": "#0048ba", "banana yellow": "#ffe135", "beaver": "#9f8170",
+                 "battleship gray": "#848482", "blueviolet": "#8a2be2", "cameo pink": "#efbbcc", "canary": "#ffff99",
+                 "candy pink": "#e4717a", "dark brown": "#654321"}
+
```
Comment (Luke-hawkins) [inline prac_05/hex_colours.py:9]: I like the please haha 10/10 :) maybe just a .lower just in case?

File: prac_05/word_occurrences.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
+"""
+Word Occurrences
+Estimate: 30 minutes
+Actual:   11 minutes
+"""
```
Comment (Luke-hawkins) [inline prac_05/word_occurrences.py:5]: Nice Time :) it all looks good to me 👍

File: prac_05/emails.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    while email != "":
+        name = extract_name(email)
+        choice = input(f"Is your name {name}? (Y/n)").upper()
+        if choice == "" or choice == "Y":
+            email_to_name[email] = name
+        else:
+            actual_name = input("Name: ")
+            email_to_name[email] = actual_name
+        email = input("Gimme your email: ")
+        extract_name(email)
```
Comment (Luke-hawkins) [inline prac_05/emails.py]: I don't think calling this extract_name function is doing anything here.

File: prac_05/emails.py [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+    print("\n")
+    for email, name in email_to_name.items():
+        print(f"{name} ({email})")
+
+
+def extract_name(email):
```
Comment (Luke-hawkins) [inline prac_05/emails.py:27]: Nice use of a comment but don't forget docstrings :)

File: prac_05/wimbledon.py [hunk 6]
```diff
@@ -0,0 +1,48 @@
+"""
+Wimbledon
+Estimate: 50 minutes
+Actual:   30 minutes
+"""
+
```
Comment (Luke-hawkins) [inline prac_05/wimbledon.py:6]: looks good to me, just some docstrings maybe 💯

**Review by Luke-hawkins — COMMENTED** [review]
Nice work 👍 🌮

**Comment by Dav0-12** [comment]
Thank you for the feedback Luke, it's been really helpful 👍 🌮

**Pull request merged**